src/EventListener/CreateSchemaListener.php

```diff
@@ -113,6 +113,14 @@ public function postGenerateSchemaTable(GenerateSchemaTableEventArgs $eventArgs)
         $revIndexName = $this->config->getRevisionFieldName().'_'.md5($revisionTable->getName()).'_idx';
         $revisionTable->addIndex([$this->config->getRevisionFieldName()], $revIndexName);
 
+        if (!$this->config->areForeignKeysDisabled()) {
+            $this->createForeignKeys($revisionTable, $revisionsTable);
```

Comment:
You now create ForeignKey before association.

I think it's better to move the `foreach ($cm->associationMappings as $associationMapping) {` inside  a 
```
if (!$this->config->areAssociationsDisabled()) {
```
to avoid impacts.